dev/src/lobster/typecheck.h

```diff
@@ -219,6 +219,22 @@ struct TypeChecker {
         }
     }
 
+    bool ConvertsToStatically(TypeRef type, TypeRef bound) {
+        if (bound == type || type->t == V_ANY)
+            return true;
+        if (bound->t != V_CLASS || type->t != V_CLASS)
+            return false;
+        if (st.SuperDistance(type->udt, bound->udt) > -1)
+            return true;
+        if (!TypeOfUndefinedGenericSpec(type))
+            return false;
+        for (auto t = bound->udt; t; t = t->resolved_superclass) {
+            if (t->first == type->udt->first)
+                return true;
```

Comment:
I was about to type a long comment here explaining why checks of `SuperDistance` here might be problematic, but it got long, so instead rewrote it as a general issue here: https://github.com/aardappel/lobster/issues/99